service/dap/daptest/client.go

```diff
@@ -120,6 +118,16 @@ func (c *Client) ExpectConfigurationDoneResponse(t *testing.T) *dap.Configuratio
 	return c.expectReadProtocolMessage(t).(*dap.ConfigurationDoneResponse)
 }
 
+func (c *Client) ExpectThreadsResponse(t *testing.T) *dap.ThreadsResponse {
+	t.Helper()
+	return c.expectReadProtocolMessage(t).(*dap.ThreadsResponse)
```

Comment:
fyi - if this type assertion fails (because c.expectReadProtoMessage returns a nil or something other type that can't be type asserted as *dap.ThreadsResponse, this can cause a runtime panic. In testing, panic is not that great. Something like this will avoid runtime panic. 
v, ok := c.expectReadProtoMessage(t).(*dap.ThreadsResponse)
if !ok {
   t.Errorf(...)
}
return v

Reply:
Originally, the idea was to keep these helper functions short because we abstracted them out. A panic in a test will fail the test and show where it failed, so it seemed acceptable. FWIW these specific failures/panics are expected to be extremely rare.

Adding a non-panicking type assertion will replace panics by errors, but it will also increase the size of the repetitive code considerably. This has to be repeated for every message type (including the error message added in `t.Errorf`), and code-generation seemed like an overkill for these test helpers.

That's the tradeoff that seemed reasonable to me at the time of writing. I don't have a strong feeling about this.

Reply:
Thanks for the clarification. Since this is only for test, maybe ok. 

I personally don't trust panic handling in test (e.g. golang.org' issue/37555 as one of the recent failures)